### PR TITLE
[CU-2v2hx9p] Move linter checks to nix

### PR DESCRIPTION
**Pull request by andor0**
## Issue
https://app.clickup.com/t/2v2hx9p

## Description
Also, I removed unused dependencies.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2v2hx9p Move linter checks from an old CI pipeline to a new CI pipeline](https://app.clickup.com/t/2v2hx9p)

**Review by vimukthi-git — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -89,6 +89,60 @@ jobs:
...
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
+        with:
+          skipPush: true
+          installCommand: |
+            nix-channel --add https://nixos.org/channels/nixpkgs-22.05-darwin nixpkgs
+            nix-channel --update
+            nix-env -iA nixpkgs.cachix
```
Comment (KaiserKarel) [inline .github/workflows/build-test-deploy.yml:117]: Can we already do this with experimental features instead of `nix-env`? @cor

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -89,6 +89,60 @@ jobs:
...
+      - name: Cargo clippy
+        run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#cargo-clippy-check --no-update-lock-file --show-trace -L
+
+      - name: Cargo deny
+        run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#cargo-deny-check --no-update-lock-file --show-trace -L
+
+      - name: Cargo udeps
+        # continue-on-error: true # Often this turns gives false positives, due to the different ways that Substrate uses features.
```
Comment (KaiserKarel) [inline .github/workflows/build-test-deploy.yml:142]:
```suggestion
        # continue-on-error: true # If udeps is reporting false-positives, it is okay to continue-on-error.
```

File: runtime/primitives/src/currency.rs [hunk 3]
```diff
@@ -201,7 +201,10 @@ impl From<u128> for CurrencyId {
 #[cfg(feature = "develop")]
 impl From<CurrencyId> for xcm::latest::Junction {
 	fn from(this: CurrencyId) -> Self {
-		xcm::latest::Junction::GeneralKey(this.encode())
+		xcm::latest::Junction::GeneralKey(sp_runtime::WeakBoundedVec::force_from(
+			this.encode(),
+			None,
+		))
```
Comment (KaiserKarel) [inline runtime/primitives/src/currency.rs:207]: Why is this changed?

Comment (andor0) [reply runtime/primitives/src/currency.rs:207]: This code is enabled by `develop` feature. 
Only udeps uses `--all-features` and catches a compilation error. But the old CI uses `continue-on-error: true` for udeps check.
Perhaps, this code was broken during migration to a new Polkadot version.

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml]: Fixed.

File: .github/workflows/build-test-deploy.yml [hunk 4]
```diff
@@ -89,6 +89,60 @@ jobs:
...
+        run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#taplo-cli-check --no-update-lock-file --show-trace -L
+
+      - name: Prettier
+        run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#prettier-check --no-update-lock-file --show-trace -L
+
+      - name: Cargo clippy
+        run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#cargo-clippy-check --no-update-lock-file --show-trace -L
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:135]: this longest run, could be separate job

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml (hunk 4 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml:135]: It makes sense. I will do it in the next PR.

**Review by vimukthi-git — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by PoisonPhang — APPROVED** [review]

**Review by Dom-Roth — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@andor0 change bors.toml. I picks jobs by names. Remove non existing jobs from list and put job you create new. So may be you can also split out clippy :)

**Comment by KaiserKarel** [comment]
@Mergifyio refresh

**Comment by andor0** [comment]
@Mergifyio refresh
